samtranslator/utils/traverse.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any, Dict, List
+
+from samtranslator.utils.resolve_actions import ResolveAction
+
+
+def traverse(template: Dict[str, Any], actions: List[ResolveAction]) -> Any:
+    """
+    Calls the correct traversal method
+
+    :param template: The template that needs modifying
+    :param actions: That actions that need to be performed to the template
+    :return: Modified Template
+
+    TODO: Add multiple strategies and choose between those with an Optional value
+    """
+    return _preorder_traverse(template, actions)
+
+
+def _execute_actions(template: Dict[str, Any], actions: List[ResolveAction]) -> Any:
```

Comment:
not sure if need this function, seems like a single loop

Reply:
My idea with this here was it would make it easier to implement with separate strategies .